ui/pages/AdvancedFilter.tsx

```diff
@@ -0,0 +1,370 @@
+      to_address_hashes_to_include: getValuesArrayFromQuery(router.query.to_address_hashes_to_include),
+      from_address_hashes_to_include: getValuesArrayFromQuery(router.query.from_address_hashes_to_include),
+      to_address_hashes_to_exclude: getValuesArrayFromQuery(router.query.to_address_hashes_to_exclude),
+      from_address_hashes_to_exclude: getValuesArrayFromQuery(router.query.from_address_hashes_to_exclude),
+    };
+  });
+
+  const [ columns, setColumns ] = React.useState<Record<ColumnsIds, boolean>>(COLUMNS_CHECKED);
+  const { data, isError, isLoading, pagination, onFilterChange, isPlaceholderData } = useQueryWithPages({
+    resourceName: 'advanced_filter',
+    filters,
+    options: {
+      placeholderData: generateListStub<'advanced_filter'>(
+        ADVANCED_FILTER_ITEM,
+        50,
+        {
+          next_page_params: {
+            block_number: 5867485,
+            internal_transaction_index: 0,
+            items_count: 50,
+            token_transfer_index: null,
+            transaction_index: 2,
+          },
+          search_params: {
+            tokens: {},
+            methods: {},
+          },
+        },
+      ),
+    },
+  });
+
+  // maybe don't need to prefetch, but on dev sepolia those requests take several seconds.
+  useApiQuery('tokens', { queryParams: { limit: '7', q: '' }, queryOptions: { refetchOnMount: false } });
+  useApiQuery('advanced_filter_methods', { queryParams: { q: '' }, queryOptions: { refetchOnMount: false } });
+
+  // const downloadCSV = React.useCallback((reCaptchaToken: string) => {
+  //   apiFetch<'advanced_filter_csv', unknown, unknown>('advanced_filter_csv', {
+  //     queryParams: { recaptcha_response: reCaptchaToken },
+  //   })
+  //     .then(() => {
+  //       toast({
+  //         title: 'Please wait',
+  //         description: 'Download will start when data is ready',
+  //         status: 'warning',
+  //       });
+  //     })
+  //     .catch(() => {
+  //       toast({
+  //         title: 'Error',
+  //         description: 'Unable to download CSV',
+  //         status: 'warning',
+  //       });
+  //     });
+  // }, [ apiFetch, toast ]);
+
+  // const handleReCaptchaChange = React.useCallback((token: string | null) => {
+  //   if (token) {
+  //     downloadCSV(token);
+  //   }
+  // }, [ downloadCSV ]);
+
+  const handleFilterChange = React.useCallback((field: keyof AdvancedFilterParams, val: unknown) => {
+    setFilters(prevState => {
+      const newState = { ...prevState };
+      // fixme
+
+      // @ts-ignore:
+      newState[field] = val;
+      onFilterChange(newState.age ? omit(newState, [ 'age_from', 'age_to' ]) : newState);
+      return newState;
+    });
+  }, [ onFilterChange ]);
+
+  const onClearFilter = React.useCallback((key: keyof AdvancedFilterParams) => () => {
+    if (key === 'methods') {
+      handleFilterChange('methods_names', undefined);
```

Comment:
but that's what we need -- clear 'methods' and 'methods_names' in that case. Because we have 2 params for one filter (we save names to filters show them correctly)
same for tokens and age filter